### PR TITLE
hunspell-dict-tok: init at 20220829

### PR DESCRIPTION
**Pull request by somasis**
###### Description of changes

A `hunspell` dictionary for the constructed language Toki Pona.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/hunspell/dictionaries.nix [hunk 1]
```diff
@@ -856,4 +856,28 @@ rec {
...
+  /* TOKI PONA */
+
+  tok = mkDict rec {
+    pname = "hunspell-dict-tok";
+    version = "20220829";
+    dictFileName = "tok";
+    readmeFile = "README.en.adoc";
+
+    src = fetchzip {
+      url = "https://github.com/somasis/${pname}/releases/download/${version}/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/hunspell/dictionaries.nix:869]:
```suggestion
      url = "https://github.com/somasis/hunspell-dict-tok/releases/download/${version}/hunspell-dict-tok-${version}.tar.gz";
```

File: pkgs/development/libraries/hunspell/dictionaries.nix [hunk 2]
```diff
@@ -856,4 +856,28 @@ rec {
...
+      sha256 = "sha256-RiAODKXPUeIcf8IFcU6Tacehq5S8GYuPTuxEiN2CXD0=";
+    };
+
+    dontBuild = true;
+
+    meta = with lib; {
+      description = "Hunspell dictionary for Toki Pona";
+      homepage = "https://github.com/somasis/${pname}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/hunspell/dictionaries.nix:877]:
```suggestion
      homepage = "https://github.com/somasis/hunspell-dict-tok";
```
Now I can easily click on the link while reviewing

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/hunspell/dictionaries.nix (hunk 1 above)
Comment (somasis) [reply pkgs/development/libraries/hunspell/dictionaries.nix:869]: Apparently I didn't even bother to test these changes to the metadata after I made them originally! The GitHub repo name is `hunspell-tok`, not `hunspell-dict-tok`, whoops...

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**